src/waterfall/waterfall.cpp

```diff
@@ -37,6 +38,13 @@ Waterfall::Waterfall(QQuickItem *parent):
     connect(_updateTimer, &QTimer::timeout, this, [&] {update();});
     _updateTimer->setSingleShot(true);
     _updateTimer->start(50);
+    connect(this, &Waterfall::mouseLeave, this, [this] {_mouseInside = false; emit mouseInsideChanged();});
+    connect(this, &Waterfall::mouseMove, this, [this] {
```

Comment:
Will this introduce unnecessary overhead?

Reply:
Where is are these mouseLeave and mouseMove signals documented? Is there a mouseEnter? Is this route really better than the qml @Williangalvani had up originally?

Reply:
He could use the function that already exist..
mouseLeave and mouseMove are signals from this class, he could set the `mouseInside` variable inside this two functions and do the emit.

This will make the waterfall class more complete as widget and will make the qml part much more simple.

Reply:
Whoops, I thought they were inherited from QQuickPaintedItem.

Reply:
I think I got it right this time.